### ISSUE TITLE
Cloudflare recaptcha error

### ISSUE THREAD

**Issue by resera** [haxball/haxball-issues#871]
I host room on server and use puppeteer to automatically launch it, but today it failed cause cloudflare asks puppeteer bot to solve recaptcha. Will this be fixed or what?

**Comment by AnddyAnddy** [comment]
That's a brillant update, probably the best that could happen at this time
What are the reason for this all of tje sudden ?

**Comment by resera** [comment]
I mean, why then headless hosts are for? to run without human interaction (my guess). Thats why we could use tokens and do not solve recaptcha by hand. 

My guess is that cloudflare was installed globally on haxball.com to prevent bots in a game, but not to prevent headless host owners to use bots to launch their rooms. So it is just a mistake that will be fixed soon?

**Comment by saviola777** [comment]
I don't think that Basro changed anything, it's just a [cloudflare thing](https://support.cloudflare.com/hc/en-us/articles/203366080-Why-do-I-see-a-captcha-or-challenge-page-Attention-Required-trying-to-visit-a-site-protected-by-Cloudflare-as-a-site-visitor-) related to the recent DDoS attacks on haxball.com

Should be gone in a couple of days at most.

**Comment by AnddyAnddy** [comment]
Oh my question was to basro, sorry for the misunderstanding

**Comment by resera** [comment]
> I don't think that Basro changed anything, it's just a [cloudflare thing](https://support.cloudflare.com/hc/en-us/articles/203366080-Why-do-I-see-a-captcha-or-challenge-page-Attention-Required-trying-to-visit-a-site-protected-by-Cloudflare-as-a-site-visitor-) related to the recent DDoS attacks on haxball.com
> 
> Should be gone in a couple of days at most.

Im not familiar with cloudflare, but is there any way to exclude routes? For example do not check browser when u try to access haxball.com/headless ?

**Comment by saviola777** [comment]
> > I don't think that Basro changed anything, it's just a [cloudflare thing](https://support.cloudflare.com/hc/en-us/articles/203366080-Why-do-I-see-a-captcha-or-challenge-page-Attention-Required-trying-to-visit-a-site-protected-by-Cloudflare-as-a-site-visitor-) related to the recent DDoS attacks on haxball.com
> > Should be gone in a couple of days at most.
> 
> Im not familiar with cloudflare, but is there any way to exclude routes? For example do not check browser when u try to access haxball.com/headless ?

Might be possible, but it would defeat the purpose of having cloudflare, because attackers would then target those routes.

**Comment by resera** [comment]
I thought that purpose of this attack is not to take down entire site but target specific rooms. So whitelisting /headless route wont do any damage. But lets wait for basro

**Comment by Wazarr94** [comment]
Yesterday in the evening (CEST) the haxball website was ddosed, it could explain the protection. It was a short DDOS so no one here noticed I guess

**Comment by basro** [comment]
A DDoS attack was targeted at HaxBall servers, Cloudflare security measures were increased because of that. They are lowered now but may go up again if another attack happens.

**Comment by akelaa** [comment]
@basro I don't know what are you doing but stop using this cloudflare thing. Everytime you open this thing or set it to higher security the game becomes unplayable because of lag/shaking. So stop using this.

**Comment by saviola777** [comment]
[Here](https://hosting.review/tips-tools/what-is-cloudflare/) you can read what cloudflare is all about. It is not responsible for lag/shaking and stopping to use it is a bad idea.

**Comment by pilaf** [comment]
> I don't know what are you doing but stop using this cloudflare thing. Everytime you open this thing or set it to higher security the game becomes unplayable because of lag/shaking. So stop using this.

This comment is the equivalent of ancient people believing human sacrifices will keep the volcano god from getting mad.
